File: resources/src/Matroos.Resources/Classes/Commands/Utilities/CommandType.cs
using Matroos.Resources.Attributes;
using Matroos.Resources.Classes.Commands.CustomCommands;

namespace Matroos.Resources.Classes.Commands;

public enum CommandType
{
    [Command(typeof(MessageCommand))]
    MESSAGE,    // 0
    [Command(typeof(PingCommand))]
    PING,       // 1
    [Command(typeof(StatusCommand))]
    STATUS,     // 2
    [Command(typeof(TimerCommand))]
    TIMER,      // 3
    [Command(typeof(VersionCommand))]
    VERSION     // 4
}

File: frontend/src/utils/commandType.ts
import { CommandType } from './types';

export function CommandTypeToString(type: CommandType): string {
    switch (type) {
        case CommandType.MESSAGE:
            return 'MESSAGE';
        case CommandType.PING:
            return 'PING';
        case CommandType.STATUS:
            return 'STATUS';
        case CommandType.TIMER:
            return 'TIMER';
        case CommandType.VERSION:
            return 'VERSION';
        default:
            return '';
    }
}
